tests/src/Forms/ClientFormTest.php

```diff
@@ -4,16 +4,72 @@
 
 namespace SimpleSAML\Test\Module\oidc\Forms;
 
-use SimpleSAML\Module\oidc\Forms\ClientForm;
 use PHPUnit\Framework\TestCase;
+use SimpleSAML\Module\oidc\Forms\ClientForm;
+use SimpleSAML\Module\oidc\Forms\Controls\CsrfProtection;
+use SimpleSAML\Module\oidc\ModuleConfig;
 
 /**
  * @covers \SimpleSAML\Module\oidc\Forms\ClientForm
  */
 class ClientFormTest extends TestCase
 {
-    public function testIncomplete(): never
+    /** @var CsrfProtection */
+    protected CsrfProtection $csrfProtection;
+    /**
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $_SERVER['REQUEST_URI'] = '/';
+        $this->csrfProtection =  $this->getMockBuilder(CsrfProtection::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    public static function validateOriginProvider(): array
     {
-        $this->markTestIncomplete();
+        return [
+            ['https://example.com', true],
+            ['https://example.com.', true],
```

Comment:
I never realized you could leave the '.' in, but sure enough ` curl -iv 'https://google.com./'` connects to Google which then redirects to not having the '.'  .    Anyhow, for the same rational as my next comment, I think it is fine to have.

Reply:
done